### PR TITLE
[MOS-737] Fix for adding a country code prefix to existing contact

### PR DESCRIPTION
**Pull request by rrandomsky**
Adding or removing country code from existing contacts number is changing now number record itself, exactly when changed number is placed in DB, instead to create new number record, to avoid miss match during matching numbers.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible
- [x] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

**Comment by rrandomsky** [comment]
Smoke test passed. 
Waiting for internal package tests.

File: module-services/service-db/include/service-db/DBServiceAPI.hpp [hunk 1]
```diff
@@ -92,9 +92,12 @@ class DBServiceAPI
      *
      * @param serv - calling service
      * @param numberView - number to match contact with
+     * @param contactIDToOmit - for this contact ID the match will be ignored
      * @return std::unique_ptr<ContactRecord>
      */
-    [[deprecated]] static auto MatchContactByPhoneNumber(sys::Service *serv, const utils::PhoneNumber::View &numberView)
+    [[deprecated]] static auto MatchContactByPhoneNumber(sys::Service *serv,
+                                                         const utils::PhoneNumber::View &numberView,
+                                                         const uint32_t contactIDToOmit = 0u)
```
Comment (Maciej-Mudita) [inline module-services/service-db/include/service-db/DBServiceAPI.hpp:100]:
```suggestion
                                                         const std::uint32_t contactIDToOmit = 0u)
```

File: module-utils/phonenumber/tests/unittest_numbermatcher.cpp [hunk 2]
```diff
@@ -16,15 +16,26 @@ using namespace utils;
 class DummyHolder
 {
     PhoneNumber number;
+    static uint32_t dummyHolderCount; // init by 0
```
Comment (Maciej-Mudita) [inline module-utils/phonenumber/tests/unittest_numbermatcher.cpp:19]:
```suggestion
    static std::uint32_t dummyHolderCount; // init by 0
```

File: module-utils/phonenumber/tests/unittest_numbermatcher.cpp [hunk 3]
```diff
@@ -16,15 +16,26 @@ using namespace utils;
...
+    uint32_t contactID;
```
Comment (Maciej-Mudita) [inline module-utils/phonenumber/tests/unittest_numbermatcher.cpp:20]:
```suggestion
    std::uint32_t contactID;
```

File: module-utils/phonenumber/tests/unittest_numbermatcher.cpp [hunk 4]
```diff
@@ -16,15 +16,26 @@ using namespace utils;
...
 
   public:
-    DummyHolder(const PhoneNumber &number) : number(number)
-    {}
+    DummyHolder(const PhoneNumber &number, const uint32_t contactID = 0u) : number(number), contactID(contactID)
```
Comment (Maciej-Mudita) [inline module-utils/phonenumber/tests/unittest_numbermatcher.cpp:23]:
```suggestion
    DummyHolder(const PhoneNumber &number, const std::uint32_t contactID = 0u) : number(number), contactID(contactID)
```

File: module-utils/phonenumber/tests/unittest_numbermatcher.cpp [hunk 5]
```diff
@@ -16,15 +16,26 @@ using namespace utils;
...
     {
         return number;
     }
+
+    const decltype(contactID) &getContactID() const
+    {
+        return contactID;
+    }
 };
+uint32_t DummyHolder::dummyHolderCount = 0u;
```
Comment (Maciej-Mudita) [inline module-utils/phonenumber/tests/unittest_numbermatcher.cpp:38]:
```suggestion
std::uint32_t DummyHolder::dummyHolderCount = 0u;
```

File: module-services/service-db/include/service-db/DBContactMessage.hpp [hunk 6]
```diff
@@ -61,6 +61,41 @@ class DBContactNumberMessage : public sys::DataMessage
...
+class DBMatchContactNumberBesidesOfContactIDMessage : public sys::DataMessage
+{
+  public:
+    /**
+     * @brief Construct a new DBMatchContactNumberBesidesOfContactIDMessage object
+     *
+     * @param numberView - a number to match with
+     */
+
+    DBMatchContactNumberBesidesOfContactIDMessage(const utils::PhoneNumber::View &numberView, uint32_t contactIDToOmit);
```
Comment (Maciej-Mudita) [inline module-services/service-db/include/service-db/DBContactMessage.hpp:79]:
```suggestion
    DBMatchContactNumberBesidesOfContactIDMessage(const utils::PhoneNumber::View &numberView, std::uint32_t contactIDToOmit);
```

File: module-services/service-db/DBServiceAPI.cpp [hunk 7]
```diff
@@ -68,10 +68,11 @@ auto DBServiceAPI::ContactGetBySpeeddial(sys::Service *serv, UTF8 speeddial)
     return ContactGetByIDCommon(serv, msg);
 }
 
-auto DBServiceAPI::MatchContactByPhoneNumber(sys::Service *serv, const utils::PhoneNumber::View &numberView)
-    -> std::unique_ptr<ContactRecord>
+auto DBServiceAPI::MatchContactByPhoneNumber(sys::Service *serv,
+                                             const utils::PhoneNumber::View &numberView,
+                                             const uint32_t contactIDToOmit) -> std::unique_ptr<ContactRecord>
```
Comment (Maciej-Mudita) [inline module-services/service-db/DBServiceAPI.cpp:73]:
```suggestion
                                             const std::uint32_t contactIDToOmit) -> std::unique_ptr<ContactRecord>
```

**Review by Maciej-Mudita — APPROVED** [review]

**Review by kkleczkowski — APPROVED** [review]

**Pull request merged**